src/utils/staking.js

```diff
@@ -0,0 +1,363 @@
+import * as nearApiJs from 'near-api-js'
+import BN from 'bn.js'
+import { toNear, nearTo, gtZero } from './amounts'
+import { queryExplorer } from './explorer-api'
+
+/********************************
+Notes on deploying lockup via CLI
+
+/core-contracts/lockup/res
+near deploy --accountId=lu1.testnet --wasmFile lockup_contract.wasm --initFunction=new --initArgs='{"owner_account_id":"multisig.testnet","lockup_duration": "259200000000000","transfers_information":{"TransfersDisabled":{"transfer_poll_account_id":"vote.f863973.m0"}},"release_duration":"2592000000000000","staking_pool_whitelist_account_id":"whitelist.f863973.m0"}'
+
+near deploy --accountId=lu2.testnet --wasmFile lockup_contract.wasm --initFunction=new --initArgs='{"owner_account_id":"xa4.testnet","lockup_duration": "259200000000000","transfers_information":{"TransfersDisabled":{"transfer_poll_account_id":"vote.f863973.m0"}},"release_duration":"2592000000000000","staking_pool_whitelist_account_id":"whitelist.f863973.m0"}'
+********************************/
+
+const {
+    transactions: {
+        functionCall
+    }
+} =  nearApiJs
+const GAS_STAKE = '200000000000000' // 200 Tgas
+const GAS_2FA_STAKE = '125000000000000' // 125 Tgas for most methods
+
+// TODO Contract Helper gas amount or attached gas?
+
+const stakingMethods = {
+	viewMethods: [
+		'get_account_staked_balance',
+		'get_account_unstaked_balance',
+		'get_account_total_balance',
+		'is_account_unstaked_balance_available',
+		'get_total_staked_balance',
+		'get_owner_id',
+		'get_reward_fee_fraction',
+	],
+	changeMethods: [
+		'ping',
+        'deposit',
+        'deposit_and_stake',
+        'deposit_to_staking_pool',
+		'stake',
+		'unstake',
+		'withdraw',
+	],
+}
+
+const lockupMethods = {
+    viewMethods: [
+        'get_owners_balance',
+        'get_staking_pool_account_id',
+        'get_known_deposited_balance',
+    ]
+}
+
+const testLockup = {
+    'multisig.testnet': 'lu1.testnet', //matt2.lockup.m0
+    'xa4.testnet': 'lu2.testnet',
+}
+
+const getLockupId = async (accountId) => {
+    const hash = await crypto.subtle.digest('SHA-256', new TextEncoder().encode(accountId))
```

Comment:
probably best to just use `js-sha256` as when retrieving balance, it gets included by near-api-js anyway

Reply:
Thanks! will use!